core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/Primitive.kt

```diff
@@ -66,10 +65,42 @@ class ArrayPrimitive<T : ValidType>(collection: Collection<TypeExpression<out T>
     },
 )
 
+class ObjectPrimitive(
+    private vararg val entries: ObjectEntry<out ValidType>,
+) : Primitive<ObjectType>(
+    { manager: DopeQueryManager ->
+        val entryDopeQueries = entries.map { it.toDopeQuery(manager) }
+        DopeQuery(
+            queryString = "{${entryDopeQueries.joinToString(", ") { it.queryString }}}",
+            parameters = entryDopeQueries.fold(emptyMap()) { parameters, it -> parameters + it.parameters },
+        )
+    },
+)
+
+fun <K, V> Map<K, V>.toDopeType(): ObjectPrimitive =
```

Comment:
Here it is not possible, because of embedded Objects. In a `when` we can't assert a Generic so we have to use `is Map<*, *> -> toDopeType()` and when we want to call this `toDopeType` we don't have the Generic String as `K`. 
I hope you understand :)

Reply:
Yes, I think I got it. Thanks! 🙂

Reply:
done